internal/pkg/agent/vault/vault_file_notwindows.go

```diff
@@ -15,14 +16,12 @@ import (
 	"path/filepath"
 	"syscall"
 
-	"golang.org/x/crypto/pbkdf2"
-
 	"github.com/elastic/elastic-agent/internal/pkg/agent/vault/aesgcm"
 	"github.com/elastic/elastic-agent/pkg/utils"
 )
 
 const (
-	saltSize = 8
+	saltSize int = 128 / 8
```

Comment:
This gave an error message of `crypto/pbkdf2: use of salts shorter than 128 bits is not allowed in FIPS 140-only mode` if it was not changed.

This occurs in the crypto/pbkdf2 when it compares `len(salt) < 128/8`.

If we don't change to go 1.24.0, we will need to make this change for the FIPS distributions, i'm not sure if this is a change we can easily make to existing agents (as far as upgrade paths go)